### ISSUE TITLE
Backend allows a duplicate resource id to be added to the registry with a different type

### ISSUE THREAD

**Issue by garethsb** [bbc/nmos-registration#2]
The etcd-based implementation uses type and id as a composite key. See https://github.com/bbc/nmos-registration/blob/master/nmosregistration/etcd_backend.py.

This means that a POST to /resource with a duplicate resource id for a different type is currently accepted.

Please tell me we agree these UUIDs need to be truly unique, not unique per type!

Thanks to @rhastie for discovering this issue.

**Comment by andrewbonney** [comment]
Yes that does seem like a bug. I doubt we've ever attempted to validate that so far.
